src/reader.cpp

```diff
@@ -507,9 +507,9 @@ bool Reader::searchSuggestionsSmart(const string& prefix,
          current != suggestions.end();
          current++) {
       std::vector<std::string> suggestion;
-      suggestion.push_back(current->getTitle());
-      suggestion.push_back(current->getPath());
-      suggestion.push_back(kiwix::normalize(current->getTitle()));
+      suggestion.push_back(current.getTitle());
+      suggestion.push_back(current.getPath());
+      suggestion.push_back(kiwix::normalize(current.getTitle()));
```

Comment:
Why this change ?
Does `current` have change type ?

Reply:
I've got my answer, you use directly the `getTitle` of the iterator instead of getting a pointer to the entry and then call `getTitle`. We are good here.